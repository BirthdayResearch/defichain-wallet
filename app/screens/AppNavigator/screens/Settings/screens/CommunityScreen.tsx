import { MaterialCommunityIcons } from '@expo/vector-icons'
import * as React from 'react'
import { useCallback } from 'react'
import { FlatList, Linking, TouchableOpacity } from 'react-native'
import { Text, View } from '../../../../../components'
import { tailwind } from '../../../../../tailwind'
import { translate } from '../../../../../translations'

export function CommunityScreen (): JSX.Element {
  return (
    <FlatList
      style={tailwind('bg-gray-100')}
      data={Communities}
      ListHeaderComponent={
        <Text style={tailwind('font-medium pt-6 pb-4 px-4')}>
          {translate('screens/CommunityScreen', 'Connect with the community:')}
        </Text>
      }
      ListFooterComponent={
        <View style={tailwind('items-center py-6 px-4')}>
          <Text style={tailwind('text-gray-400 text-xs font-normal')}>
            {translate('screens/CommunityScreen', 'DeFiChain is a community-driven and open project.')}
          </Text>
          <Text style={tailwind('text-gray-400 text-xs font-normal')}>
            {translate('screens/CommunityScreen', 'The DeFiChain Foundation does not provide direct support.')}
          </Text>
        </View>
      }
      ItemSeparatorComponent={
        () => <View style={tailwind('h-px bg-gray-100')} />
      }
      renderItem={({ item }) => (
        <CommunityItemRow key={item.id} {...item} />
      )}
      testID='community_flat_list'
    />
  )
}

interface CommunityItem {
  id: string
  title: string
  url: string
  icon: React.ComponentProps<typeof MaterialCommunityIcons>['name']
}

const Communities: CommunityItem[] = [
  { id: 'gh', title: 'Report an issue on Github', url: 'https://github.com/DeFiCh/wallet/issues', icon: 'github' },
  { id: 'faq', title: 'Frequently Asked Questions', url: 'https://defichain.com/learn/#faq', icon: 'help-circle' },
  { id: 'tg_en', title: 'Telegram (EN)', url: 'https://t.me/defiblockchain', icon: 'telegram' },
  { id: 'tg_de', title: 'Telegram (DE)', url: 'https://t.me/defiblockchain_DE', icon: 'telegram' },
  { id: 'wechat', title: 'WeChat', url: 'http://weixin.qq.com/r/0xz07DzEdmEJrXiP90nB', icon: 'wechat' },
  { id: 'reddit', title: 'Reddit Community', url: 'https://www.reddit.com/r/defiblockchain/', icon: 'reddit' },
  { id: 'website', title: 'Official Website', url: 'https://defichain.com/', icon: 'web' }
]

function CommunityItemRow ({ id, title, url, icon }: CommunityItem): JSX.Element {
  const handlePress = useCallback(async () => {
    const supported = await Linking.canOpenURL(url)
    if (supported) {
      await Linking.openURL(url)
    }
  }, [url])

  return (
    <TouchableOpacity
      style={tailwind('flex-row bg-white p-4 items-center')}
      onPress={handlePress}
      testID={id}
    >
      <MaterialCommunityIcons name={icon} size={24} style={tailwind('text-primary')} />
      <Text style={tailwind('ml-2')}>
        {translate('screens/CommunityScreen', title)}
      </Text>
    </TouchableOpacity>
  )
}
