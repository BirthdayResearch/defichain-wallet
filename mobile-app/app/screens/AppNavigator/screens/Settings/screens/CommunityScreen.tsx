import { MaterialCommunityIcons } from '@expo/vector-icons'
import * as React from 'react'
import { View } from 'react-native'
import {
  ThemedFlatList,
  ThemedIcon,
  ThemedSectionTitle,
  ThemedText,
  ThemedTouchableOpacity,
  ThemedView
} from '@components/themed'
import { tailwind } from '@tailwind'
import { translate } from '@translations'
import { openURL } from '@api/linking'

export function CommunityScreen (): JSX.Element {
  return (
    <ThemedFlatList
      ItemSeparatorComponent={
        () => (
          <ThemedView
            dark={tailwind('bg-gray-700')}
            light={tailwind('bg-gray-100')}
            style={tailwind('h-px')}
          />
        )
      }
      ListHeaderComponent={
        <ThemedSectionTitle
          testID='community_title'
          text={translate('screens/CommunityScreen', 'CONNECT WITH THE COMMUNITY')}
        />
      }
      data={Communities}
      renderItem={({ item }) => (
        <CommunityItemRow
          key={item.id}
          {...item}
        />
      )}
      testID='community_flat_list'
    />
  )
}

interface CommunityItem {
  id: string
  title: string
  url: string
  icon: React.ComponentProps<typeof MaterialCommunityIcons>['name']
}

const Communities: CommunityItem[] = [
  {
    id: 'gh',
    title: 'Report an issue on Github',
    url: 'https://github.com/DeFiCh/wallet/issues',
    icon: 'github'
  },
  {
    id: 'faq',
    title: 'Frequently Asked Questions',
    url: 'https://defichain.com/learn/#faq',
    icon: 'help-circle'
  },
  {
    id: 'tg_en',
    title: 'Telegram (EN)',
    url: 'https://t.me/defiblockchain',
    icon: 'telegram'
  },
  {
    id: 'tg_de',
    title: 'Telegram (DE)',
    url: 'https://t.me/defiblockchain_DE',
    icon: 'telegram'
  },
  {
    id: 'announcements',
    title: 'Announcements',
    url: 'https://t.me/defichain_announcements',
    icon: 'telegram'
  },
  {
    id: 'wechat',
    title: 'WeChat',
    url: 'http://weixin.qq.com/r/0xz07DzEdmEJrXiP90nB',
    icon: 'wechat'
  }
]

function CommunityItemRow ({
  id,
  title,
  url,
  icon
}: CommunityItem): JSX.Element {
  const handlePress = async (): Promise<void> => {
    await openURL(url)
  }

  return (
    <ThemedTouchableOpacity
      onPress={handlePress}
      style={tailwind('flex-row p-4 items-center')}
      testID={id}
    >
      <ThemedIcon
        dark={tailwind('text-darkprimary-500')}
        iconType='MaterialCommunityIcons'
        light={tailwind('text-primary-500')}
        name={icon}
        size={24}
      />

      <ThemedText style={tailwind('ml-4 font-medium')}>
        {translate('screens/CommunityScreen', title)}
      </ThemedText>

      <View
        style={tailwind('flex flex-grow justify-end items-end')}
      >
        <ThemedIcon
          dark={tailwind('text-gray-200')}
          iconType='MaterialIcons'
          light={tailwind('text-gray-500')}
          name='chevron-right'
          size={24}
        />
      </View>
    </ThemedTouchableOpacity>
  )
}
