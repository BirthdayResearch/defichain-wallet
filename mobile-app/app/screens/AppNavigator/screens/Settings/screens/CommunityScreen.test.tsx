import { fireEvent, render } from '@testing-library/react-native'
import { Linking } from 'react-native'
import { CommunityScreen } from './CommunityScreen'

jest.mock('@shared-contexts/ThemeProvider')

it('<CommunityScreen /> should match snapshot', async () => {
  const tree = render(<CommunityScreen />)
  expect(tree.toJSON()).toMatchSnapshot()
  const receiveButton = await tree.findByTestId('gh')
  fireEvent.press(receiveButton)
  expect(Linking.canOpenURL).toBeCalled()
})
