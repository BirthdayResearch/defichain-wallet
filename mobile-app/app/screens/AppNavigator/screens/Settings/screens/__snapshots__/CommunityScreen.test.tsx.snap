// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommunityScreen /> should match snapshot 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListHeaderComponent={
    <ThemedSectionTitle
      testID="community_title"
      text="CONNECT WITH THE COMMUNITY"
    />
  }
  data={
    Array [
      Object {
        "icon": "github",
        "id": "gh",
        "title": "Report an issue on Github",
        "url": "https://github.com/DeFiCh/wallet/issues",
      },
      Object {
        "icon": "help-circle",
        "id": "faq",
        "title": "Frequently Asked Questions",
        "url": "https://defichain.com/learn/#faq",
      },
      Object {
        "icon": "telegram",
        "id": "tg_en",
        "title": "Telegram (EN)",
        "url": "https://t.me/defiblockchain",
      },
      Object {
        "icon": "telegram",
        "id": "tg_de",
        "title": "Telegram (DE)",
        "url": "https://t.me/defiblockchain_DE",
      },
      Object {
        "icon": "telegram",
        "id": "announcements",
        "title": "Announcements",
        "url": "https://t.me/defichain_announcements",
      },
      Object {
        "icon": "wechat",
        "id": "wechat",
        "title": "WeChat",
        "url": "http://weixin.qq.com/r/0xz07DzEdmEJrXiP90nB",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="community_flat_list"
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Array [
              Object {
                "color": "rgba(115, 115, 115, 1)",
                "fontFamily": "MediumFont",
                "fontSize": 12,
                "fontWeight": "500",
                "lineHeight": 16,
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 24,
              },
              Object {
                "backgroundColor": "rgba(245, 245, 245, 1)",
                "color": "rgba(115, 115, 115, 1)",
              },
            ],
          ]
        }
        testID="community_title"
        text="CONNECT WITH THE COMMUNITY"
      >
        CONNECT WITH THE COMMUNITY
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderBottomColor": "rgba(229, 229, 229, 1)",
            "borderBottomWidth": 1,
            "borderLeftColor": "rgba(229, 229, 229, 1)",
            "borderRightColor": "rgba(229, 229, 229, 1)",
            "borderTopColor": "rgba(229, 229, 229, 1)",
            "flexDirection": "row",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="gh"
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Report an issue on Github
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1,
            },
            Object {
              "backgroundColor": "rgba(245, 245, 245, 1)",
            },
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderBottomColor": "rgba(229, 229, 229, 1)",
            "borderBottomWidth": 1,
            "borderLeftColor": "rgba(229, 229, 229, 1)",
            "borderRightColor": "rgba(229, 229, 229, 1)",
            "borderTopColor": "rgba(229, 229, 229, 1)",
            "flexDirection": "row",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="faq"
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Frequently Asked Questions
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1,
            },
            Object {
              "backgroundColor": "rgba(245, 245, 245, 1)",
            },
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderBottomColor": "rgba(229, 229, 229, 1)",
            "borderBottomWidth": 1,
            "borderLeftColor": "rgba(229, 229, 229, 1)",
            "borderRightColor": "rgba(229, 229, 229, 1)",
            "borderTopColor": "rgba(229, 229, 229, 1)",
            "flexDirection": "row",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="tg_en"
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Telegram (EN)
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1,
            },
            Object {
              "backgroundColor": "rgba(245, 245, 245, 1)",
            },
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderBottomColor": "rgba(229, 229, 229, 1)",
            "borderBottomWidth": 1,
            "borderLeftColor": "rgba(229, 229, 229, 1)",
            "borderRightColor": "rgba(229, 229, 229, 1)",
            "borderTopColor": "rgba(229, 229, 229, 1)",
            "flexDirection": "row",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="tg_de"
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Telegram (DE)
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1,
            },
            Object {
              "backgroundColor": "rgba(245, 245, 245, 1)",
            },
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderBottomColor": "rgba(229, 229, 229, 1)",
            "borderBottomWidth": 1,
            "borderLeftColor": "rgba(229, 229, 229, 1)",
            "borderRightColor": "rgba(229, 229, 229, 1)",
            "borderTopColor": "rgba(229, 229, 229, 1)",
            "flexDirection": "row",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="announcements"
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          Announcements
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1,
            },
            Object {
              "backgroundColor": "rgba(245, 245, 245, 1)",
            },
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderBottomColor": "rgba(229, 229, 229, 1)",
            "borderBottomWidth": 1,
            "borderLeftColor": "rgba(229, 229, 229, 1)",
            "borderRightColor": "rgba(229, 229, 229, 1)",
            "borderTopColor": "rgba(229, 229, 229, 1)",
            "flexDirection": "row",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="wechat"
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontWeight": "500",
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                },
              ],
            ]
          }
        >
          WeChat
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
